src/controller/src/controller.cpp

```diff
@@ -74,4 +74,5 @@ void Controller::press_button(Button button) { set_button(button, true); }
 
 void Controller::release_button(Button button) { set_button(button, false); }
 
+void Controller::tick() {}
```

Comment:
Why not add the ticking logic in `controller.cpp`, and just leave it unimplemented in `RecordingController`